### ISSUE TITLE
cas oauth call back fails with cas 5.1.0-rc3

### ISSUE THREAD

**Issue by mrcasablr** [davido/gerrit-oauth-provider#87]
Hello,

I'm using CAS 5.1.0-RC3, Gerrit 2.13.6 and cas-oauth-plugin for 2.13.6.  When I click "signin" in gerrit UI, I get redirected to CAS SSO login page.  On successful authentication, the redirect back to gerrit fails with "Application Not Authorized to Use CAS"

The callback URL seem to be `https://gerrit.xyz.com:8443/oauth`  Is this correct? Below are my configs and the error in detaiil

Here is my gerrit config for oauth
```
[plugin "gerrit-oauth-provider-cas-oauth"]
        root-url = https://sso.xyz.com:8443/cas
        client-id = gerrit
        fix-legacy-user-id = false
```
Here is my CAS oauth service definition
```
{
  "@class" : "org.apereo.cas.support.oauth.services.OAuthRegisteredService",
  "clientId": "gerrit",
  "clientSecret": "***",
  "bypassApprovalPrompt": false,
  "generateRefreshToken": false,
  "serviceId" : "^https://gerrit.xyz.com:8444.*",
  "name" : "Gerrit",
  "id" : 30000,
  "attributeReleasePolicy" : {
    "@class" : "org.apereo.cas.services.ReturnAllAttributeReleasePolicy"
  }
}
```
Here is the relevant log snippet from the cas.log
```
...
2017-04-08 05:35:15,417 DEBUG [org.apereo.cas.authentication.PseudoPlatformTransactionManager] - <Initiating transaction commit>
2017-04-08 05:35:15,418 DEBUG [org.apereo.cas.authentication.PseudoPlatformTransactionManager] - <Resuming suspended transaction after completion of inner transaction>
2017-04-08 05:35:15,418 DEBUG [org.apereo.cas.authentication.PseudoPlatformTransactionManager] - <Initiating transaction commit>
2017-04-08 05:35:15,438 DEBUG [org.apereo.cas.support.oauth.validator.OAuth20Validator] - <[client_id]: [gerrit]>
2017-04-08 05:35:15,438 DEBUG [org.apereo.cas.support.oauth.validator.OAuth20Validator] - <[redirect_uri]: [https://gerrit.xyz.com:8444/oauth]>
2017-04-08 05:35:15,439 DEBUG [org.apereo.cas.support.oauth.validator.OAuth20Validator] - <[response_type]: [null]>
2017-04-08 05:35:15,439 ERROR [org.apereo.cas.support.oauth.validator.OAuth20Validator] - <Missing: [response_type]>
2017-04-08 05:35:15,440 ERROR [org.apereo.cas.support.oauth.web.endpoints.OAuth20AuthorizeEndpointController] - <Authorize request verification failed>
2017-04-08 05:35:15,445 DEBUG [org.apereo.cas.authentication.principal.WebApplicationServiceFactory] - <No service is specified in the request. Skipping service creation>
2017-04-08 05:35:15,445 DEBUG [org.apereo.cas.web.support.DefaultArgumentExtractor] - <No service could be extracted based on the given request>
2017-04-08 05:35:15,445 DEBUG [org.apereo.cas.web.support.AbstractArgumentExtractor] - <Extractor did not generate service.>
2017-04-08 05:35:15,453 ERROR [org.apache.velocity] - <ResourceManager : unable to find resource 'casServiceErrorView.vm' in any resource loader.>
...
```

**Comment by mwebber** [comment]
Possibly the same problem as issue #92

**Comment by mwebber** [comment]
@mrcasablr, we think this problem has been fixed (see issue #92). When the next release is made, please re-test, and close this issue if it is working.

**Comment by davido** [comment]
The release is here: https://github.com/davido/gerrit-oauth-provider/releases/tag/v2.14.2 .
